### PR TITLE
doc: improve electrophoresis sample.

### PR DESCRIPTION
**Pull request by KaiSzuttor**
- added observable version for persistence length analysis and corrected python analysis
- improved sampling

old plot shows wrong raw data:

persistence_length_old

new plot:

persistence_length

### CODE REVIEW

File: samples/electrophoresis.py
Comment (jngrad) [inline samples/electrophoresis.py]: you could drop the conditional with `for i in range(1, N_MONOMERS):`

File: testsuite/scripts/samples/test_electrophoresis.py
Comment (jngrad) [inline testsuite/scripts/samples/test_electrophoresis.py]: The error message
```
  File "testsuite/scripts/samples/test_electrophoresis.py", line 38, in test_mobility
    "mobility = {:.2f}".format(sample.mu))
AssertionError: False is not true : mobility = 10.26
```
is not very helpful, you may consider instead:
```python
self.assertAlmostEqual(sample.mu, 10.2, delta=0.1)
```
which outputs
```
  File "testsuite/scripts/samples/test_electrophoresis.py", line 37, in test_mobility
    self.assertAlmostEqual(sample.mu, 10.2, delta=0.1)
AssertionError: 10.341812051847997 != 10.2 within 0.1 delta
```

File: testsuite/scripts/samples/test_electrophoresis.py
Comment (jngrad) [inline testsuite/scripts/samples/test_electrophoresis.py]:
```suggestion
                               msg="persistence length of {:.1f} not correct".format(sample.persistence_length))
```
I don't think the `msg` argument was designed with this kind of code duplication in mind, which is prone to subtle copy-paste errors that will actually hurt maintainers instead of guiding them. The default error message of unittest assertions is already clear enough, showing both the function name (`test_persistence_length`) and the faulty variable if the assertion is a one-liner. I would vote for removing custom messages that only format the variable names. I'm not even sure it can be automated via `inspect` in a reliable way (see https://stackoverflow.com/a/2749881, https://stackoverflow.com/a/2749857).

**Review by jngrad — CHANGES_REQUESTED** [review]

File: testsuite/scripts/samples/test_electrophoresis.py
Comment (KaiSzuttor) [reply testsuite/scripts/samples/test_electrophoresis.py]: I don't have a strong opinion on that  but I also think that we should not over-engineer one-liners.

**Review by jngrad — APPROVED** [review]
pending CI

**Comment by jngrad** [comment]
I'm not following: did kodiak just merge `python` into a PR whose last commit did not pass CI?

**Comment by KaiSzuttor** [comment]
it did not wait for the CI to pass because this PR is the only one in the queue

**Comment by jngrad** [comment]
I thought kodiak would not merge `python` if the GitHub status said CI failed.

**Comment by KaiSzuttor** [comment]
``` bash
# if there are running status checks on a PR when it's up for merge, don't
# wait for those to finish before updating the branch
optimistic_updates = false # default: true
```
https://kodiakhq.com/docs/detailed-setup

**Comment by jngrad** [comment]
Ah right, thanks!

**Pull request merged**
